src/buildercore/threadbare/__init__.py

```diff
@@ -0,0 +1,19 @@
+# gevent is used by parallel-ssh which interferes with Python multiprocessing and futures
+# it can cause indefinite blocking.
+# this bit of magic appears to make everything work nicely with each other.
+# - http://www.gevent.org/api/gevent.monkey.html
+from gevent import monkey
+
+monkey.patch_all()
```

Comment:
I've [had this failing before I think](https://github.com/elifesciences/elife-spectrum/pull/44#issuecomment-297313791), because of a library that was doing it when it shouldn't, or wasn't doing it when it should have. Keep an eye on it during rollout.

Reply:
Parallelism in Python is a shambles if you ask me and likely to cause a headache later. I had to remove xdist because of it.

I'm eyeing ParallelSSH and wondering if we can't fork it and strip out all of the parallel stuff. It was originally using Paramiko until they developed their own backend called Python-SSH and now they offer either backend to use, so there is a Paramiko dependency it's pulling in as well. Another headache and maintenance burden I don't want.

I was using Python-SSH to begin with but it was too low level and I didn't want to be messing about with sockets and control codes.